EKF/vel_pos_fusion.cpp

```diff
@@ -119,7 +120,7 @@ void Ekf::fuseVelPosHeight()
 	}
 
 	if (_fuse_height) {
-		if (_control_status.flags.baro_hgt) {
+		if (_control_status.flags.baro_hgt && !_control_status.flags.gps && !_control_status.flags.ev_pos && !(_control_status.flags.rng_hgt && (_R_to_earth(2, 2) > 0.7071f))) {
```

Comment:
Nowhere in control.cpp or ekf.cpp is the logic for setting "flags.baro_hgt = false" available yet when using external vision sensor for height. This, is if(baro height) else if(vision height) fails. 

The solution above gives higher priority to external vision height updates  when available over baro height, even though both "_control_status.flags.baro_hgt" and "_control_status.flags.ev_pos" are true.  This is tested to work, otherwise height updates from external  vision were neglected. 

A nice addition would be to reset the "_control_status.flags.baro_hgt" when external vision height is available.